model.py

```diff
@@ -72,10 +72,8 @@ def update(self, input_pos, k_val, v_val):
         # input_pos: [S], k_val: [B, H, S, D]
         assert input_pos.shape[0] == k_val.shape[2]
 
-        k_out = self.k_cache
-        v_out = self.v_cache
-        k_out[:, :, input_pos] = k_val
-        v_out[:, :, input_pos] = v_val
+        k_out = torch.ops.aten.index_put_(self.k_cache, [None, None, input_pos], k_val)
```

Comment:
Why is this change needed?

Reply:
dynamo.export does not support capturing mutation (the mutated tensor will be written to a new tensor) I think, this is added to work around that, @HDCharles have more context on this